### PR TITLE
Add newer versions of server to compatibility matrix

### LINKED ISSUE #629: [BUG] Compatibility.md does not include latest versions of OpenSearch
### What is the bug?
Coming from https://forum.opensearch.org/t/java-client-compatibility/15917.

https://github.com/opensearch-project/opensearch-java/blob/main/COMPATIBILITY.md shows compatibility of the latest client up to core v2.6, whereas OpenSearch is on the verge of releasing 2.10.

Can the compatibility.md get updated?

Maybe a script is needed to create an issue when the server bumps, or a script that checks the api specification to see if there are uncovered versions in compatibility.md?

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
Same as title

### Issues Resolved
Resolves #629 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/test-integration.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ jobs:
           - { opensearch_version: 2.5.0, java: 11 }
           - { opensearch_version: 2.6.0, java: 11 }
           - { opensearch_version: 2.7.0, java: 11 }
+          - { opensearch_version: 2.8.0, java: 11 }
+          - { opensearch_version: 2.9.0, java: 11 }
```
Comment (reta) [inline .github/workflows/test-integration.yml:25]: Just fyi, 2.10 is supposed to be released on Monday (:crossed_fingers: ), may be we could include it as well

**Review by reta — DISMISSED** [review]

File: .github/workflows/test-integration.yml (hunk 1 above)
Comment (VachaShah) [reply .github/workflows/test-integration.yml:25]: Definitely! I will leave this open till then.

**Comment by VachaShah** [comment]
The compatibility with 2.9 is failing, I will take a look into it.

**Comment by VachaShah** [comment]
With 2.9.0 server, seeing the following error in integration tests:

```
Caused by:
        org.opensearch.client.ResponseException: method [DELETE], host [https://localhost:9200], URI [/.plugins-ml-config], status line [HTTP/1.1 403 Forbidden]
        {"error":{"root_cause":[{"type":"security_exception","reason":"no permissions for [] and User [name=admin, backend_roles=[admin], requestedTenant=null]"}],"type":"security_exception","reason":"no permissions for [] and User [name=admin, backend_roles=[admin], requestedTenant=null]"},"status":403}
            at app//org.opensearch.client.transport.rest_client.RestClientTransport.getHighLevelResponse(RestClientTransport.java:256)
            ... 41 more

**Comment by VachaShah** [comment]
> With 2.9.0 server, seeing the following error in integration tests:
> 
> ```
> Caused by:
>         org.opensearch.client.ResponseException: method [DELETE], host [https://localhost:9200], URI [/.plugins-ml-config], status line [HTTP/1.1 403 Forbidden]
>         {"error":{"root_cause":[{"type":"security_exception","reason":"no permissions for [] and User [name=admin, backend_roles=[admin], requestedTenant=null]"}],"type":"security_exception","reason":"no permissions for [] and User [name=admin, backend_roles=[admin], requestedTenant=null]"},"status":403}
>             at app//org.opensearch.client.transport.rest_client.RestClientTransport.getHighLevelResponse(RestClientTransport.java:256)
>             ... 41 more
> ```

This is fixed in the latest commits in this PR.

**Review by dblock — APPROVED** [review]

**Pull request merged**